src/conn.c

```diff
@@ -66,14 +66,14 @@ static conn_registry_t *acquire_registry(conn_mode_entry_t *entry, udp_socket_co
 		registry = calloc(1, sizeof(conn_registry_t));
 		if (!registry) {
 			JLOG_FATAL("Memory allocation failed for connections registry");
-			return NULL;
+			return (void*)-1;
```

Comment:
I'm not a fan of cast -1 to a pointer here. Could you please change the function to `int acquire_registry(conn_mode_entry_t *entry, udp_socket_config_t *config)` (which would return -1 on error and 0 on success) and then access the registry with `entry->registry` instead?